### ISSUE TITLE
[Issue]: Current version not set

### ISSUE THREAD

**Issue by jerone** [coreybutler/nvm-windows#1198]
### What happened?

Running `nvm use 22.12.0` won't set the current version.

![image](https://github.com/user-attachments/assets/c8b4603c-a214-41d6-a2b6-7553fe358c4a)

I had the previous version installed. After running the v1.2.0 installer, Node.js won't work anymore. I uninstalled NVM4W and removed any Node.js reference I could find. Re-installed v1.2.0 and run `nvm install lts` and `nvm use 22.12.0` and it still didn't set the current version.

I tried different prompts/shells: PowerShell for Windows, PowerShell 7, Command Prompt, Bash.

I also tried to run the installer as Administrator.

I verified that Environment Variables are set.
![image](https://github.com/user-attachments/assets/269af3c7-2277-4c0a-8ef8-aaa968ef3f8d)

Then I noticed that there was also no folder `nvm4w` under `C:\`.
After creating that folder and manually creating the [symlink](https://github.com/coreybutler/nvm-windows/blob/de4ee40711b2a966ad8c8bd543e16bdcd68eae65/src/nvm.go#L1160), I finally had a working current version.
`C:\nvm4w>mklink /D "nodejs" "C:\Users\jeron\AppData\Roaming\nvm\v22.12.0"`
![image](https://github.com/user-attachments/assets/687e149f-7ccc-454a-bbb8-c302f7264991)
_(Probably creating the folder was enough, and adding the symlink was redundant)_

**Maybe the new `C:\nvm4w\` directory is not created during installation and symlink fails silently.**

### What did you expect to happen?

NVM use should set the current version.

### Version

1.2.0 or newer (Default)

### Which version of Windows?

Windows 10

### Which locale?

English (Default)

### Which shell are you running NVM4W in?

Command Prompt, PowerShell

### User Permissions?

Standard User, Elevated

### Is Developer Mode enabled?

No (Default)

### Debug Output

```Shell
jeron is not using admin or elevated rights.

Windows Version:        10.0 (Build 19045)

Windows Developer Mode: UNKNOWN

NVM4W Version:          1.2.0
NVM4W Path:             C:\Users\jeron\AppData\Roaming\nvm\nvm.exe
NVM4W Settings:         C:\Users\jeron\AppData\Roaming\nvm\settings.txt
NVM_HOME:               C:\Users\jeron\AppData\Roaming\nvm
NVM_SYMLINK:            C:\nvm4w\nodejs
Node Installations:     C:\Users\jeron\AppData\Roaming\nvm
Default Architecture:   64-bit
Mirrors:                No mirrors configured
HTTP Proxy:             none

Total Node.js Versions: 2
Active Node.js Version: none
(run "nvm use <version>" to activate a version)
NVM_SYMLINK does not exist yet. This is auto-created when "nvm use" is run.

PROBLEMS DETECTED
-----------------
"pwsh in jeron" not recognized: the Command Prompt and Powershell are the only officially supported consoles. Some features may not work as expected.
```

### Anything else?

Based on `nvm debug` results it appears that PowerShell (and also Command Prompt) is not correctly recognized within the Windows Terminal program.

**Comment by coreybutler** [comment]
This does indeed look like the installer is not creating the `C:\nvm4w` directory. The temporary fix is to create that directory manually, then run `nvm use lts` again.

**Comment by thadguidry** [comment]
@coreybutler On Windows 11, If I use the new 1.2.0 installer to use install into a new path `E:\nvm` and the symlink to `E:\nvm\nodejs` where are downloads actually getting installed to 😄 notice it downloaded `lts` into...where??

```
PS C:\WINDOWS\system32> nvm install lts
Downloading node.js version 22.12.0 (64-bit)...
Extracting node and npm...
Complete

Installation complete.
If you want to use this version, type:

nvm use 22.12.0
PS C:\WINDOWS\system32> nvm list

No installations recognized.
PS C:\WINDOWS\system32> nvm current
No current version. Run 'nvm use x.x.x' to set a version.
PS C:\WINDOWS\system32> nvm use 22.12.0
activation error: Version not installed. Run "nvm ls" to see available versions.
PS C:\WINDOWS\system32> nvm ls

No installations recognized.
PS C:\WINDOWS\system32>
```
![image](https://github.com/user-attachments/assets/034c822f-5dd2-421d-b645-99fd77807419)

**Comment by coreybutler** [comment]
@thadguidry run `nvm debug` to see where your installs are being saved. Here's my current one as a reference:

![image](https://github.com/user-attachments/assets/c672d7bd-c561-4208-b556-6c38861c7026)

**Comment by thadguidry** [comment]
Powershell Non-Admin shows:

```PS C:\WINDOWS\system32> nvm debug
thadg is not using admin or elevated rights.

Windows Version:        10.0 (Build 26100)

Windows Developer Mode: UNKNOWN

NVM4W Version:          1.2.0
NVM4W Path:             E:\nvm\nvm.exe
NVM4W Settings:         E:\nvm\settings.txt
NVM_HOME:               E:\nvm
NVM_SYMLINK:            C:\Program Files\nodejs
Node Installations:     E:\nvm
Default Architecture:   64-bit
Mirrors:                No mirrors configured
HTTP Proxy:             none

Total Node.js Versions: 0
Active Node.js Version: none
(run "nvm use <version>" to activate a version)
panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xc0000005 code=0x0 addr=0x28 pc=0x3f4670]

goroutine 1 [running]:
main.checkLocalEnvironment()
        D:/a/nvm-windows/nvm-windows/src/nvm.go:1572 +0x1950
main.main()
        D:/a/nvm-windows/nvm-windows/src/nvm.go:283 +0x645
PS C:\WINDOWS\system32>
```

Powershell in Admin mode shows:
```
PowerShell 7.4.6
PS C:\Users\thadg> nvm debug
Running NVM for Windows with administrator privileges.

Windows Version:        10.0 (Build 26100)

NVM4W Version:          1.2.0
NVM4W Path:             E:\nvm\nvm.exe
NVM4W Settings:         E:\nvm\settings.txt
NVM_HOME:               E:\nvm
NVM_SYMLINK:            E:\nvm\nodejs
Node Installations:     E:\nvm
Default Architecture:   64-bit
Mirrors:                No mirrors configured
HTTP Proxy:             none

Total Node.js Versions: 0
Active Node.js Version: none
(run "nvm use <version>" to activate a version)
NVM_SYMLINK does not exist yet. This is auto-created when "nvm use" is run.

IPv6 is enabled. This has been known to slow downloads significantly.

No problems detected.

⚠️  We are currently building Author/Runtime to succeed NVM for Windows.

⚠️  An upgrade is available: v1.2.1
   run "nvm upgrade" to update.


Find help at https://github.com/coreybutler/nvm-windows/wiki/Common-Issues
PS C:\Users\thadg>
```

**Comment by thadguidry** [comment]
Ah!!!, but a NEW PS Terminal now shows correctly in non-admin mode!
```
PowerShell 7.4.6
PS C:\Users\thadg> nvm debug
thadg is not using admin or elevated rights.

Windows Version:        10.0 (Build 26100)

Windows Developer Mode: UNKNOWN

NVM4W Version:          1.2.0
NVM4W Path:             E:\nvm\nvm.exe
NVM4W Settings:         E:\nvm\settings.txt
NVM_HOME:               E:\nvm
NVM_SYMLINK:            E:\nvm\nodejs
Node Installations:     E:\nvm
Default Architecture:   64-bit
Mirrors:                No mirrors configured
HTTP Proxy:             none

Total Node.js Versions: 0
Active Node.js Version: none
(run "nvm use <version>" to activate a version)
NVM_SYMLINK does not exist yet. This is auto-created when "nvm use" is run.

IPv6 is enabled. This has been known to slow downloads significantly.

No problems detected.

⚠️  We are currently building Author/Runtime to succeed NVM for Windows.

⚠️  An upgrade is available: v1.2.1
   run "nvm upgrade" to update.


Find help at https://github.com/coreybutler/nvm-windows/wiki/Common-Issues
PS C:\Users\thadg> nvm upgrade
Installation canceled by user
PS C:\Users\thadg>
```

So it seems like there needs to be a note on the last page of nvm-setup about 
"IMPORTANT! Ensure that you start a NEW terminal after installing `nvm-setup.exe`, otherwise environment settings will NOT be read correctly."
Let's add that to the last page in the installer?

**Comment by coreybutler** [comment]
@thadguidry I'm more perplexed by the fact you installed the lts version and it didn't show up in `E:\NVM`.

Did you open the original powershell prompt using the installer (i.e. the checkbox on the last screen)?

**Comment by thadguidry** [comment]
Scratch the Let's add a note... seems there's more to the issue at hand:
even with a NEW PS window non-admin... debug shows that `NVM_SYMLINK:  E:\nvm\nodejs`  but yet, it doesn't seem to install under that symlink?  Even if I manually create the `E:\nvm\nodejs` subfolder... and try again... `nvm install lts` it still seems to download and install somewhere???
```
PS C:\Users\thadg> nvm install lts
Downloading node.js version 22.12.0 (64-bit)...
Extracting node and npm...
Complete

Installation complete.
If you want to use this version, type:

nvm use 22.12.0
PS C:\Users\thadg>
```

I am opening brand NEW terminals.. I've closed ALL terminals and just launched both a NEW PS as admin and without... same thing happens where a `nvm install lts` seems to download successfully but darn if I can figure out where/how?

**Comment by thadguidry** [comment]
I have even removed the `nodejs` subfolder symlink under `C:\Program Files\` just to see what `nvm debug` then shows which at least now it shows there's an invalid symlink issue...so that's a "good thing".
```
PS C:\Users\thadg> nvm debug
thadg is not using admin or elevated rights.

Windows Version:        10.0 (Build 26100)

Windows Developer Mode: UNKNOWN

NVM4W Version:          1.2.0
NVM4W Path:             E:\nvm\nvm.exe
NVM4W Settings:         E:\nvm\settings.txt
NVM_HOME:               E:\nvm
NVM_SYMLINK:            E:\nvm\nodejs
Node Installations:     E:\nvm
Default Architecture:   64-bit
Mirrors:                No mirrors configured
HTTP Proxy:             none

Total Node.js Versions: 0
Active Node.js Version: none
(run "nvm use <version>" to activate a version)

IPv6 is enabled. This has been known to slow downloads significantly.

PROBLEMS DETECTED
-----------------
NVM_SYMLINK (E:\nvm\nodejs) is not a valid symlink.


⚠️  We are currently building Author/Runtime to succeed NVM for Windows.


Find help at https://github.com/coreybutler/nvm-windows/wiki/Common-Issues
PS C:\Users\thadg>
```

**Comment by thadguidry** [comment]
Reinstalling `nvm-setup.exe` 1.2.0 doesn't seem to create the symlink back in place either?

**Comment by thadguidry** [comment]
Look at how crazy this flow seems?

```
PS C:\Users\thadg> nvm debug
thadg is not using admin or elevated rights.

Windows Version:        10.0 (Build 26100)

Windows Developer Mode: UNKNOWN

NVM4W Version:          1.2.0
NVM4W Path:             E:\nvm\nvm.exe
NVM4W Settings:         E:\nvm\settings.txt
NVM_HOME:               E:\nvm
NVM_SYMLINK:            E:\nvm\nodejs
Node Installations:     E:\nvm
Default Architecture:   64-bit
Mirrors:                No mirrors configured
HTTP Proxy:             none

Total Node.js Versions: 0
Active Node.js Version: none
(run "nvm use <version>" to activate a version)
NVM_SYMLINK does not exist yet. This is auto-created when "nvm use" is run.

IPv6 is enabled. This has been known to slow downloads significantly.

No problems detected.

⚠️  We are currently building Author/Runtime to succeed NVM for Windows.


Find help at https://github.com/coreybutler/nvm-windows/wiki/Common-Issues
PS C:\Users\thadg> nvm use lts
activation error: Version not installed. Run "nvm ls" to see available versions.
PS C:\Users\thadg> nvm install lts
Downloading node.js version 22.12.0 (64-bit)...
Extracting node and npm...
Complete

Installation complete.
If you want to use this version, type:

nvm use 22.12.0
PS C:\Users\thadg> nvm use lts
activation error: Version not installed. Run "nvm ls" to see available versions.
PS C:\Users\thadg> nvm use 22.12.0
activation error: Version not installed. Run "nvm ls" to see available versions.
PS C:\Users\thadg>
```

**Comment by thadguidry** [comment]
> Did you open the original powershell prompt using the installer (i.e. the checkbox on the last screen)?

YES, that's the very first output I showed you at the beginning.  Doesn't matter.. even with closing and reopening a new terminal, we can see that the ENV's are seemingly getting picked up by `nvm debug` and yet, they are not being used.

**Comment by coreybutler** [comment]
@thadguidry do you have an antivirus program running or do you have protected directories enabled? The download process creates a temporary directory and is supposed to move the files to the node install location. It _looks_ like that is happening, but obviously it is not. An error would be thrown if there were a problem, so I'm wondering if AV/protected directories are removing the files right after they're copied to the destination directory (`E:\nvm` in your case).

**Comment by thadguidry** [comment]
Possible... only Windows Defender... let me check...

**Comment by thadguidry** [comment]
nothing in Windows Defender has been blocked or quarantined.
![image](https://github.com/user-attachments/assets/31cb05e6-51d0-4e0f-951f-bd85f58172bc)

**Comment by thadguidry** [comment]
Shall I pull down source, compile in Go and debug or output logging to help ya'll troubleshoot?

**Comment by coreybutler** [comment]
If you have time to do that, it would be phenomenal, because I cannot recreate the error. I'm currently working on some other patches (the update system had some code that got copied into the release when it should not have). Look for the `web.GetNodeJS` method.

I'll try to put together a debugging version tomorrow. If you do find the source, or even additional info, I'd be very appreciative.

**Comment by coreybutler** [comment]
I forgot to mention, we have a tool for building/running the app called [qgo](https://github.com/quikdev/go) (QuikGo). Once installed, you `cd ./src` and run `qgo build` or `qgo build --no-cache`.

**Comment by vmasek** [comment]
I had a same problem, creating `C:\nvm4w` manually and running `nvm use 22` again solved my problem 👍🏻

**Comment by thadguidry** [comment]
@coreybutler what moves or links the files under the `root` TEMP into the `NYM_SYMLINK` or perhaps `NVM_HOME\nodejs\` ?
I just don't see that happening, but indeed the download and zip extraction happens just fine.
I've added a few DEBUG println statements into the code to help see a bit more and indeed all the Node files are there extracted nicely in the `root` TEMP folder:
```
PS F:\GitHubRepos\nvm-windows\bin> ./nvm.exe install lts
DEBUG arch.Validate:  C:\Users\thadg\AppData\Local\Temp\nvm-install-2291855044 22.12.0 64 false
DEBUG URL: https://nodejs.org/dist/v22.12.0/node-v22.12.0-win-x64.zip
Downloading node.js version 22.12.0 (64-bit)... 
Extracting node and npm...
DEBUG Extracting node and npm:  <nil>
DEBUG zip:  C:\Users\thadg\AppData\Local\Temp\nvm-install-2291855044\v22.12.0\node-v22.12.0-win-x64
DEBUG root:  C:\Users\thadg\AppData\Local\Temp\nvm-install-2291855044\v22.12.0
DEBUG fs.Move: <nil>
Complete

Installation complete.
If you want to use this version, type:

nvm use 22.12.0
```
If I do `nvm use 22.12.0` is there supposed to be some file movement?
![image](https://github.com/user-attachments/assets/f87b3677-7e09-47b1-abcb-28178991ad84)

**Comment by thadguidry** [comment]
And I've ensured that there's a correctly made (I made it manually just to be sure) symlink
```
PS F:\GitHubRepos\nvm-windows\bin> ./nvm.exe debug
thadg is not using admin or elevated rights.

Windows Version:        10.0 (Build 26100)

Windows Developer Mode: UNKNOWN

NVM4W Version:          1.2.1
NVM4W Author Bridge:    Not Detected
NVM4W Path:             E:\nvm\nvm.exe
NVM4W Settings:         E:\nvm\settings.txt
NVM_HOME:               E:\nvm
NVM_SYMLINK:            C:\Program Files\nodejs
Node Installations:     E:\nvm
Default Architecture:   64-bit
Mirrors:                No mirrors configured
HTTP Proxy:             none

Total Node.js Versions: 0
Active Node.js Version: none
(run "nvm use <version>" to activate a version)

IPv6 is enabled. This has been known to slow downloads significantly.

PROBLEMS DETECTED
-----------------
The author-nvm.exe file is missing or not executable.


⚠️  We are currently building Author/Runtime to succeed NVM for Windows.


Find help at https://github.com/coreybutler/nvm-windows/wiki/Common-Issues
```

![image](https://github.com/user-attachments/assets/9ffe5caa-db52-4ad6-911c-216e643913b3)

**Comment by coreybutler** [comment]
@thadguidry thank you for this.

I've been able to add a debugging mode for the install process, which I'll release in v1.2.1. I based some of the logs off of what you provided. That said, I think this conversation has veered off from the original post. I'll close this with the commit for v1.2.1. If you are still struggling with this after updating to the new version, open a new thread. Your issue is related to the download process whereas this issue is related to the linking process and the installer.

@jerone this is resolved and will be closed/available in v1.2.1 (later today, CST). Unfortunately, there was also an issue (#1199) with the upgrade capability, so it will require a reinstall.

**Comment by valpioner** [comment]
I had the same issue today when `nvm use` didn't do anything, after spending a few hours, it was resolved by manually creating folder `C:/nvm4w` because nvm won't make it automatically.

**Comment by coreybutler** [comment]
@thadguidry I've released 1.2.1. It will require a reinstall, but now you can run `nvm install lts --verbose` and it will output a full detailed log of everything it's doing. If that still doesn't work, open a new issue.

**Comment by jerone** [comment]
@coreybutler Thanks. Confirmed.

The installation process, when NVM4W is already installed, could use a little friendly improvement:
![image](https://github.com/user-attachments/assets/99252d12-d9f3-422e-923b-b47f8bdacec7)
I had to manually remove the folder that NVM4W (should) create itself.

**Comment by coreybutler** [comment]
@jerone - I saw that too and was annoyed by it ;-) I added a fix in the 1.2.1 uninstaller, but I can't change it in 1.2.0.
